### DISCUSSION TITLE
Use Unicode CLDR for translations of language, territory, calendar, etc. names

### DISCUSSION THREAD

**Discussion by bact** [Laravel-Lang/common#169] [Ideas]
### Feature description

The Unicode Common Locale Data Repository (CLDR) project provides "Translations of names: languages, scripts, countries and regions, currencies, eras, months, weekdays, day periods, time zones, cities, and time units, emoji characters and sequences"

https://cldr.unicode.org/

Localization community can utilize that for faster and more standardized translations.

**Comment by bact** [comment]
For example, core.zip from http://unicode.org/Public/cldr/44/
has common/main/ that contains translations of language, territory, calendar names in more than 100 languages.

**Comment by andrey-helldar** [comment]
1. https://github.com/Laravel-Lang/native-locale-names
2. https://github.com/Laravel-Lang/native-country-names
3. https://github.com/Laravel-Lang/native-currency-names

🙂

1. https://laravel-lang.com/usage-native-locales.html
2. https://laravel-lang.com/usage-facade.html#examples


As for the `laravel-lang/lang` package, I haven't yet figured out how to easily add translations of country names to it, since [Punic](https://punic.readthedocs.io/), which prepares the data from CLDR, gets it by localization code. But they can be converted for English too, so it's impossible to make a list of matches in advance.

**Reply by andrey-helldar** [reply]
On the one hand we can add translation for many values from CLDR, but on the other hand, this may not be very useful from the end developer's point of view because we need to use query keys for translation. This means that we won't be able to apply such values to translate Laravel Nova files, for example. Thus, the project will essentially have two types of translation - one "boxed" and the other "custom".

For example:

Option 1:
```
lang/
  fr/
    countries.php
  fr.json
```
```php
__('France'); // From Laravel Nova
__('countries.fr'); // From out package
```

Option 2:
```
lang/
  fr.json
```
```php
__('France'); // From Laravel Nova
__('fr'); // From out package
```

**Reply by andrey-helldar** [reply]
If you have an idea on how to apply translations from CLDR to Laravel Nova, that would be great!

Why Laravel Nova? It's simple - it has the [largest file by number of translation strings](https://github.com/Laravel-Lang/lang/blob/main/source/nova/4.x/nova.json), including country names and months.

**Reply by andrey-helldar** [reply]
There is an idea in key mapping, but it is similar to working with regular expressions - very long, slow and high chance of detecting changes in the keys themselves.

For example:

```php
// locale => translation key
$map = [
    'France' => 'fr',
];

return CLDR->translate($map['France'] ?? 'France')
```

Why I haven't done it yet - very many countries in the file.
